packages/next/src/server/app-render/action-handler.ts

```diff
@@ -253,28 +253,41 @@ export async function handleAction({
           const {
             decodeReply,
             decodeReplyFromBusboy,
+            decodeAction,
           } = require(`react-server-dom-webpack/server.node`)
 
           if (isMultipartAction) {
-            const busboy = require('busboy')
-            const bb = busboy({ headers: req.headers })
-            req.pipe(bb)
+            if (isFetchAction) {
+              const busboy = require('busboy')
+              const bb = busboy({ headers: req.headers })
+              req.pipe(bb)
 
-            bound = await decodeReplyFromBusboy(bb, serverModuleMap)
+              bound = await decodeReplyFromBusboy(bb, serverModuleMap)
+            } else {
+              // React doesn't yet publish a busboy version of decodeAction
+              // so we polyfill the parsing of FormData.
+              const { Readable } = require('stream')
+              const UndiciRequest = require('undici').Request
```

Comment:
This is an unfortunate hack. I'm not sure if there's a better way to get these imports.

I'll probably just add a `decodeActionFromBusboy` for Node.js in React.

Reply:
For `Request`, this is better as we have `edge-runtime` imported already:

```
const { Request } = require('next/dist/compiled/@edge-runtime/primitives/fetch')
```

Reply:
If use `undici`, we need to change it to `require('next/dist/compiled/undici')` (there's also a lint error for this).